test/types/helper_types.test-d.ts

```diff
@@ -0,0 +1,72 @@
+import { expectAssignable, expectNotType, expectType } from 'tsd';
+import type {
+  AcceptedFields,
+  KeysOfAType,
+  KeysOfOtherType,
+  NotAcceptedFields,
+  NumericType,
+  ObjectFilterOperators,
+  DotAndArrayNotation,
+  OnlyFieldsOfType,
+  IntegerType,
+  IsAny
+} from '../../src/mongo_types';
+import { Decimal128, Double, Int32, Long, Document } from '../../src/index';
+
+expectType<IsAny<any, true, false>>(true);
+expectNotType<IsAny<number, true, false>>(true);
+expectNotType<IsAny<true, true, false>>(true);
+expectNotType<IsAny<false, true, false>>(true);
+expectNotType<IsAny<0, true, false>>(true);
+expectNotType<IsAny<never, true, false>>(true);
+expectNotType<IsAny<unknown, true, false>>(true);
+expectNotType<IsAny<any[], true, false>>(true);
+
+expectAssignable<IntegerType>(2);
+expectAssignable<IntegerType>(2.3); // Typescript cannot assert anything about this
+expectNotType<IntegerType>(new Double(2));
+
+expectAssignable<NumericType>(2);
+expectAssignable<NumericType>(Long.fromInt(3));
+expectAssignable<NumericType>(new Decimal128('2.3'));
+expectAssignable<NumericType>(new Int32(23));
+expectAssignable<NumericType>(new Double(2.3));
+
+// ObjectFilterOperators
+expectAssignable<ObjectFilterOperators<{ a: number }>>({ a: { $eq: 2 } });
+
+// KeysOfAType - union of keys that are the second type argument
+expectType<KeysOfAType<{ a: number; b: string }, number>>('a');
```

Comment:
I am a little bit confused about the intent here, I think it's just the plurality of the naming - if KeysOfAType is really supposed to only apply to a single variable, wouldn't it make more sense to call it Key and not Keys? same question for the "KeysOfOtherType" helper